### DISCUSSION TITLE
Jenkins pipeline

### DISCUSSION THREAD

**Discussion by aju2372** [gravitational/teleport#27722] [Q&A]
Hi team. 
I have a question i have added my Jenkins server and Jenkins UI behind my teleport and it is working fine. But my concern is that when i run a Jenkins job, the job collects the updated packages and push the packages to pypi server which is also behind teleport. I followed the document "Using Teleport Machine ID with Jenkins" but at last the command which is mentioned in this document i am bit confused by this
 "steps {
  sh "ssh -F /opt/machine-id/ssh_config root@node-name.example.com hostname"
}" 
when my job is ready to push the packages to pypi server which is behind teleport it first hits the ulr of the pypi server and then it should be authenticated to teleport and after that it pushes the packages to pypi server.
can you please help me how can i do this. And where i have crater the bot on teleport server or on jenkins server.

**Comment by webvictim** [comment]
I think you'd want to configure Jenkins to connect to your Pypi server behind Teleport using a Machine ID bot - see this guide: https://goteleport.com/docs/machine-id/guides/applications/
